code/Core/BackgroundTraffic.py

```diff
@@ -20,56 +20,53 @@ def __init__(self, pcap_dest_path):
-                str_tcp_seg = str(ip_payload.payload)
-
-                if http_pkt and ip_payload.haslayer(TCP) and ip_payload.getfieldval("dport") == http_port and len(str_tcp_seg) == payload_size:
-                    payload = http_pkt[Raw].load.decode()
-                    http_pkt[Raw].load = payload
-                    if accept_encoding_for_malicious_attack_packet in payload:
-                        final_attack_packet = pkt_num + 1
-                        victim_ip = ip_pkt.getfieldval("dst")
-                        return self.modify_background_traffic(victim_ip, final_attack_packet)
+                pkt_metadata = pkt[1]
+                ip_pkt = eth_frame.payload
+                if pkt_num == 12: 
+                    victim_ip = ip_pkt.getfieldval("dst")
+                    timestamp_for_malicious_packet = pkt_metadata.sec + pkt_metadata.usec / 1e6
+                if pkt_num == 13: 
+                    background_manipulation = False
+            return victim_ip, timestamp_for_malicious_packet, background_manipulation
 
-        def modify_background_traffic(self, victim_ip, final_attack_packet):
+        def modify_background_traffic(self, victim_ip, timestamp_for_malicious_packet):
             """
             Deletes all activity the IP address of the victim after having received 
             the malicious attack packet.
+
             :return updated pcap file.
             """
             input_pcap_raw_packets = scapy.utils.RawPcapReader(self.pcap_dest_path)
-            for pkt_num, pkt in enumerate(input_pcap_raw_packets):
+            for _, pkt in enumerate(input_pcap_raw_packets):
                 eth_frame = inet.Ether(pkt[0])
+                pkt_metadata = pkt[1]
                 ip_pkt = eth_frame.payload
                 ip_payload = ip_pkt.payload
                 http_pkt = ip_payload.payload
+                timestamp = pkt_metadata.sec + pkt_metadata.usec / 1e6
                 if http_pkt: 
                     ip_src = ip_pkt.getfieldval("src")
                     ip_dst = ip_pkt.getfieldval("dst")
-                    if pkt_num < final_attack_packet:
+                    if timestamp < timestamp_for_malicious_packet:
                         new_pkt = new_pkt = (eth_frame / ip_pkt / ip_payload)
                         self.packets.append(new_pkt)
-                    if pkt_num > final_attack_packet and ip_src != victim_ip and ip_dst != victim_ip:
+                    if timestamp > timestamp_for_malicious_packet and ip_src != victim_ip and ip_dst != victim_ip:
                         new_pkt = new_pkt = (eth_frame / ip_pkt / ip_payload)
                         self.packets.append(new_pkt)
                 else: 
-                    if pkt_num < final_attack_packet: 
+                    if timestamp < timestamp_for_malicious_packet: 
                         new_pkt = new_pkt = (eth_frame / ip_pkt / ip_payload)
                         self.packets.append(new_pkt)
```

Comment:
The if statements here seem to be doing mostly the same thing. For example, the if statement in line 69 and if statement in line 62. What we want to do is to bg remove traffic after injection for exploited victim, all else traffic is fine. Below code checks if traffic is before injection time then, it is added to bg. If after injection time and it does not belong to victim then it is also added. Otherwise we move on to next packet. 
```suggestion
                ip_src = ip_pkt.getfieldval("src")
                ip_dst = ip_pkt.getfieldval("dst")
                is_victim_ip = ip_src == victim_ip or ip_dst == victim_ip
                if timestamp < timestamp_for_malicious_packet or (timestamp > timestamp_for_malicious_packet and not is_victim_ip):
                    new_pkt = self.create_new_packet(eth_frame, ip_pkt, ip_payload)
                    self.packets.append(new_pkt)
```